source/TimePicker.js

```diff
@@ -585,14 +602,32 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
+			var meridiems = event.originator.get('meridiems'),
+				oldMeridiem = meridiems[event.old],
```

Comment:
Rather than updating the properties of the `onChange` event fired by `IntegerPicker`, I wonder if we can just encapsulate this logic inside `MeridiemPicker` whenever its value changes (we would have access to the previous value), as we should know the most recent hour and minute (via `setValueByTime`), and come up with a clean way for `TimePicker` to retrieve the hour and minute in this change handler. It seems like the logic here is very meridiem-specific.